lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_ansible_method_base.rb

```diff
@@ -120,12 +132,22 @@ def current_info
 
     def build_options_hash
       @aem.options.tap do |config_info|
-        config_info[:extra_vars] = MiqAeEngine::MiqAeReference.encode(@inputs)
+        config_info[:extra_vars] = MiqAeEngine::MiqAeReference.encode(ansible_stats_from_ws.merge(@inputs))
         config_info[:extra_vars][:manageiq] = method_manageiq_env
         config_info[:extra_vars][:manageiq_connection] = manageiq_connection_env(@workspace.ae_user)
       end
     end
 
+    def ansible_stats_from_ws
+      @workspace.persist_state_hash.each_with_object({}) do |(attr, val), obj|
+        if attr.start_with?(ANSIBLE_STATS_PREFIX)
+          obj[attr[ANSIBLE_STATS_PREFIX_LEN..-1]] = val
+        else
+          obj[attr] = val
+        end
```

Comment:
With this loop a state_var called `var1` and one called `ansible_stats_var1` will only result in one hash key being set, which is expected.  My concern is that we have no way to guarantee the order and therefore could have unpredictable results.

I would suggest we create two hashes and at the end of the method return the result of merging the `non_ansible_hash` into the `ansible_hash` which would ensure the non-Ansible keys win and gives a consistent result.

Something like: `ansible_hash.merge(non_ansible_hash)`